### PR TITLE
factor levels for barplot data

### PR DESCRIPTION
**Pull request by OliBravo**

### CODE REVIEW

File: R/contents.R [hunk 1]
```diff
@@ -90,6 +90,25 @@ getLayerGeom <- function(layer) {
...
+    factors <- Filter(
+      function(name) { is.factor(origin[[name]]) },
+      names(origin)
+    )
+    for (f in factors) {
+      if (f %in% names(df)) {
+        orig_lvl <- levels(origin[[f]])
+        df_lvl <- unique(df[[f]])
+
+        for (r in seq_len(nrow(df))) {
```
Comment (cosi1) [inline R/contents.R:103]: This
```
df[, f] <- orgi_lvl[which(df_lvl == df[, f])]
```
should work equally good, but much faster.

Comment (OliBravo) [reply R/contents.R]: This part has been completely redesigned.

File: inst/example/server.R [hunk 2]
```diff
@@ -7,13 +7,13 @@ plotIris <- function(x_aes, y_aes) {
     theme(legend.position = "bottom")
 }
 
-plotIrisBarplot <- function() {
+plotMtcarsBarplot <- function() {
+  d_fac <- mtcars
+  d_fac[, c("am", "gear")] <- lapply(d_fac[, c("am", "gear")], as.factor)
```
Comment (adamgolubowski) [inline inst/example/server.R:12]: repeated column names can be moved to variable

**Comment by cosi1** [comment]
LGTM

**Review by cosi1 — APPROVED** [review]

**Pull request merged**